### PR TITLE
BUG: Check version of crypt provider

### LINKED ISSUE #2107: Wrong cryptography providers might be loaded on Python 3.6
If `cryptography` is installed in the local environment where PyPDF is installed as well, reading password-protected PDFs will fail on Python 3.6. It seems like the `cryptography` version is just too old in this case.

The background seems to be that while the package metadata does not use `cryptography` on Python 3.6, this is not reflected in the actual imports, as can be seen in the simple interactive session or in the stacktrace itself:

https://github.com/py-pdf/pypdf/blob/89eb626a7a7e22937b9216e817f5882431196b24/pyproject.toml#L47-L50
https://github.com/py-pdf/pypdf/blob/89eb626a7a7e22937b9216e817f5882431196b24/pypdf/_crypt_providers/__init__.py#L30-L41

As `cryptography` is installed for some other packages, PyPDF will use it, although apparently incompatible.

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
Linux-4.12.14-lp151.28.91-default-x86_64-with-glibc2.3.4

$ python -c "import pypdf;print(pypdf.__version__)"
3.15.2
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
>>> from pypdf import _crypt_providers
>>> print(_crypt_providers.aes_cbc_decrypt.__module__)
'pypdf._crypt_providers._cryptography'
```

## Traceback

This is the relevant traceback I see:

```
  File "/home/user/lib/python3.6/site-packages/pypdf/_writer.py", line 204, in __init__
    clone_from = PdfReader(clone_from)
  File "/home/user/lib/python3.6/site-packages/pypdf/_reader.py", line 337, in __init__
    self._encryption.verify(pwd) == PasswordType.NOT_DECRYPTED
  File "/home/user/lib/python3.6/site-packages/pypdf/_encryption.py", line 978, in verify
    key, rc = self.verify_v4(pwd) if self.V <= 4 else self.verify_v5(pwd)
  File "/home/user/lib/python3.6/site-packages/pypdf/_encryption.py", line 994, in verify_v4
    self.EncryptMetadata,
  File "/home/user/lib/python3.6/site-packages/pypdf/_encryption.py", line 480, in verify_owner_password
    user_password = RC4_decrypt(key, user_password)
  File "/home/user/lib/python3.6/site-packages/pypdf/_crypt_providers/_cryptography.py", line 88, in rc4_decrypt
    decryptor = Cipher(ARC4(key), mode=None).decryptor()
TypeError: __init__() missing 1 required positional argument: 'backend'
```

### PR DESCRIPTION
**Pull request by MartinThoma**
Closes #2107

### CODE REVIEW

**Comment by MartinThoma** [comment]
@stefan6419846 Do you think this solves the issue?

**Comment by stefan6419846** [comment]
> Do you think this solves the issue?

Basically yes, but this introduces the external `packaging` dependency without further declaration, which might not be available for some reasons. Could we avoid this by some string comparison or similar?

File: pypdf/_crypt_providers/__init__.py [hunk 1]
```diff
@@ -77,4 +84,5 @@
     "aes_ecb_decrypt",
     "aes_cbc_encrypt",
     "aes_cbc_decrypt",
+    "crypt_provider",
```
Comment (MartinThoma) [inline pypdf/_crypt_providers/__init__.py:87]:
```suggestion
```

**Comment by MartinThoma** [comment]
You're right. I've removed the dependency.

**Pull request merged**